--- a/lib/alpn.ml
+++ b/lib/alpn.ml
@@ -134,8 +134,9 @@
   | (Some "http/1.1" | None), `V1 request ->
       let error_handler = error_handler_v1 edn error_handler in
       let response_handler = response_handler_v1_1 edn response_handler in
-      let body, conn =
-        Httpaf.Client_connection.request request ~error_handler
+      let conn = Httpaf.Client_connection.create ?config:None in
+      let body =
+        Httpaf.Client_connection.request conn request ~error_handler
           ~response_handler in
       Lwt.async (fun () ->
           Paf.run (module Httpaf_Client_connection) ~sleep conn flow) ;
